### PR TITLE
Py3 upgrade

### PR DESCRIPTION
**Pull request by whatisgalen**
re: #77

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ env:
   - UPGRADE_WEBPY=false
   - UPGRADE_WEBPY=true
 python:
-  - 2.7
```
Comment (mekarpeles) [inline .travis.yml:7]: I think we'll want to be compliant with both here

Comment (cclauss) [reply .travis.yml:7]: Yes.  2to3 often breaks Python 2 compatibility to achieve Python 3 compatibility.  It is better to use futurize or modernize instead.  These are wrappers around 2to3 that curb its most destructive tendencies.

File: infogami/core/code.py [hunk 2]
```diff
@@ -77,13 +77,13 @@ def trim(self, d):
...
         elif isinstance(d, dict):
-            for k, v in d.items():
+            for k, v in list(d.items()):
                 d[k] = self.trim(v)
                 if d[k] is None or d[k] == '' or d[k] == []:
                     del d[k]
 
             # hack to stop saving empty properties
-            if d.keys() == [] or d.keys() == ['unique']:
+            if list(d.keys()) == [] or list(d.keys()) == ['unique']:
```
Comment (cclauss) [inline infogami/core/code.py:86]:
```suggestion
            if list(d) in ([], ['unique']):
```

File: infogami/core/code.py [hunk 3]
```diff
@@ -95,11 +95,11 @@ def POST(self, path):
         i = web.storage(helpers.unflatten(i))
         i.key = path
 
-        _ = web.storage((k, i.pop(k)) for k in i.keys() if k.startswith('_'))
+        _ = web.storage((k, i.pop(k)) for k in list(i.keys()) if k.startswith('_'))
```
Comment (cclauss) [inline infogami/core/code.py:98]:
```suggestion
        _ = web.storage((k, i.pop(k)) for k in list(i) if k.startswith('_'))
```

**Comment by cclauss** [comment]
Please rebase.

**Comment by mekarpeles** [comment]
I think most of these have been applied piece-wise. Thank you for generating this manifest?

**Pull request closed without merging**